### PR TITLE
OpenVDB Points Convert SOP Fixes

### PR DESCRIPTION
**Pull request by Idclip**
- Fixed a bug where the auto voxel transform was being calculated on the packed geometry in the OpenVDB Points Convert SOP
- Fixed a bug where points representing packed geometry were also being converted in the OpenVDB Points Convert SOP

Signed-off-by: Nick Avramoussis <4256455+Idclip@users.noreply.github.com>

### CODE REVIEW

File: openvdb_houdini/SOP_OpenVDB_Points_Convert.cc [hunk 1]
```diff
@@ -799,6 +799,45 @@ SOP_OpenVDB_Points_Convert::Cache::cookVDBSop(OP_Context& context)
...
+        }
+        else {
+            // Convert the prim group to points - we have to use mergePoints
+            // instead of mergePrims to make sure we merge points that are not
+            // associated with any primitive
+            GA_PointGroup pointsWithPackedPrims(*ptGeo);
+            pointsWithPackedPrims.GA_ElementGroup::operator|=(*packgroup);
+
+            // Merge everything except the geo associated with prims we've just unpacked
+            nonConstDetail.mergePoints(*ptGeo, GA_Range(pointsWithPackedPrims, /*invert*/true));
```
Comment (Idclip) [inline openvdb_houdini/SOP_OpenVDB_Points_Convert.cc:834]: @jmlait might be worth removing the operator overrides for `GA_PointGroup`, `GA_PrimitiveGroup` etc as they seem to just call through to `GA_ElementGroup` and obfuscate the input group type - assuming I've understood this behaviour correctly!

File: openvdb_houdini/SOP_OpenVDB_Points_Convert.cc [hunk 2]
```diff
@@ -827,40 +866,11 @@ SOP_OpenVDB_Points_Convert::Cache::cookVDBSop(OP_Context& context)
...
-
-            detail = ptGeo;
-        }
-        else {
-            // merge unpacked and packed point data
-
-            nonConstDetail.mergePoints(*ptGeo);
-            detail = &nonConstDetail;
-        }
-
```
Comment (Idclip) [inline openvdb_houdini/SOP_OpenVDB_Points_Convert.cc:863]: Altered and moved this logic to *before* the calculation of the voxel size

File: openvdb_houdini/SOP_OpenVDB_Points_Convert.cc (hunk 1 above)
Comment (jmlait) [reply openvdb_houdini/SOP_OpenVDB_Points_Convert.cc:834]: The &=, -=, etc overrides make a lot of sense as you can only apply that with matching group types.  So you can't have 
pointgroup -= primgroup;
As far as I understand, the overrides then enforce that restriction.
The issue is that |= does implicit conversion of the group type.  The ElementGroup version used to be a private method until GA_ElementGroup was merged into GA_ATIGroupBool. The latter was always meant to be a private interface, but got exposed as ElementGroup must be public.  (This was 4y9m ago, so likely compatible with all current OpenVDB targets)

Long story short, the |= working with different group types wasn't meant to be exposed :>  I think the best solution is to have the GA_PointGroup overrides accept an ElementGroup as the second input.  I'll add an RFE to this effect and see if there are any issues.

**Comment by jmlait** [comment]
@danrbailey : Isn't this such a filter?

    parms.add(hutil::ParmFactory(PRM_STRING, "group", "Group")
        .setChoiceList(&hutil::PrimGroupMenu)
        .setTooltip("Specify a subset of the input point data grids to convert.")

Or do you mean to also filter the primitives to unpack?

File: openvdb_houdini/SOP_OpenVDB_Points_Convert.cc (hunk 1 above)
Comment (jmlait) [reply openvdb_houdini/SOP_OpenVDB_Points_Convert.cc:834]: Why not use combine() rather than this awkward |=?

Comment (Idclip) [reply openvdb_houdini/SOP_OpenVDB_Points_Convert.cc:834]: I wasn't entirely sure if combine did the same thing but will definitely switch if so! Is this the main way to "cast" one group type to another?

**Comment by danrbailey** [comment]
> @danrbailey : Isn't this such a filter?
> 
> ```
> parms.add(hutil::ParmFactory(PRM_STRING, "group", "Group")
>     .setChoiceList(&hutil::PrimGroupMenu)
>     .setTooltip("Specify a subset of the input point data grids to convert.")
> ```
> 
> Or do you mean to also filter the primitives to unpack?

Well, this parameter is the primitive group filter for conversion in the other direction (VDB -> Houdini). It's hidden for this direction (Houdini -> VDB) because it doesn't make a lot of sense to have a primitive group filter for points. However, it would at least make sense to include a points group filter, but whether or not that's a good idea or not, I'm not sure. I notice that some of the other VDB sops that handle points (such as VDB from particles) also don't expose a points group filter, so maybe it's just not convention to do this? I can't remember if we talked about this at some point and then decided against it.

**Comment by jmlait** [comment]
I think I decided against the same named parameter being used as different filter types.  Having a separate ptgroup filter does make sense here though.  It could also subsume some of the implicit logic that is going on, like rejecting points that are part of packed primitives.

**Comment by Idclip** [comment]
> It could also subsume some of the implicit logic that is going on, like rejecting points that are part of packed primitives.

Are you suggesting we shouldn't always be ignoring points representing packed geo? I was going to suggest we extend this behaviour for points representing other geo too (i.e. volumes). Would converting these points ever be useful/wanted?

File: openvdb_houdini/SOP_OpenVDB_Points_Convert.cc (hunk 1 above)
Comment (jmlait) [reply openvdb_houdini/SOP_OpenVDB_Points_Convert.cc:834]: Yep, combine will chain to |= if the src is an element group. It then handles in addition weird stuff like edge groups.

The |= in ElementGroup will be marked deprecated in 18.5.

**Comment by jmlait** [comment]
I think we should always ignore points of packed geo.  But it definitely is the case one could imagine other sets of points to be removed as well.  Having a group makes this easier to do without an extra blast operation.  (Which also may avoid a defragment, so be a performance win)

**Comment by Idclip** [comment]
Will merge this soon

File: openvdb_houdini/SOP_OpenVDB_Points_Convert.cc (hunk 1 above)
Comment (Idclip) [reply openvdb_houdini/SOP_OpenVDB_Points_Convert.cc:834]: Switch to combine!
